src/components/text/Subheading.jsx

```diff
@@ -0,0 +1,87 @@
+// @flow strict
+
+import * as React from 'react';
+import classNames from 'classnames';
+import {SUBHEADING_TYPE, SUBHEADING_SIZE} from './subheadingConsts';
+import type {TextColorType} from './Text';
+
+export type SubheadingTypeType =
+  | 'span'
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6';
+
+export type SubheadingSizeType =
+  | 'xsmall'
+  | 'small'
+  | 'medium'
+  | 'large'
+  | 'xlarge'
+  | 'xxlarge'
+  | 'xxxlarge';
+
+export type SubheadingTransformType = 'uppercase' | 'lowercase' | 'capitalize';
+
+export type SubheadingAlignType =
+  | 'to-left'
+  | 'to-center'
+  | 'to-right'
+  | 'justify';
+
+export {
+  SUBHEADING_TYPE,
+  SUBHEADING_SIZE,
+  SUBHEADING_TRANSFORM,
+  SUBHEADING_ALIGN,
+} from './subheadingConsts';
+
+export {TEXT_COLOR} from './Text';
+
+export type SubheadingPropsType = {
+  children?: React.Node,
+  size?: SubheadingSizeType,
+  type?: SubheadingTypeType,
+  color?: ?TextColorType,
+  transform?: ?SubheadingTransformType,
+  align?: ?SubheadingAlignType,
+  className?: ?string,
+  inherited?: ?boolean,
+  ...
+};
+
+const Subheading = ({
+  children,
+  type = SUBHEADING_TYPE.H1,
```

Comment:
This is not safe from accessibility perspective

Reply:
I agree, h2 will be better as default. 

This applies to all other text components (out of scope of this pr).

Reply:
Ok, I'm changing it to h2.

Reply:
Done